hail/src/main/scala/is/hail/expr/ir/MatrixValue.scala

```diff
@@ -35,6 +35,39 @@ case class MatrixValue(
       colValues.value.map(a => Row.fromSeq(queriers.map(_ (a))))
     }
 
+    def stringSampleIds: IndexedSeq[String] = {
+      val colKeyTypes = typ.colKeyStruct.types
+      assert(colKeyTypes.length == 1 && colKeyTypes(0).isInstanceOf[TString], colKeyTypes.toSeq)
+      val querier = typ.colType.query(typ.colKey(0))
+      colValues.value.map(querier(_).asInstanceOf[String])
+    }
+
+    def requireRowKeyVariant(method: String) {
```

Comment:
These checks are already done in Python - why do we need them in the JVM?

The python ones are way more readable (no java stack trace)